### ISSUE TITLE
twisted.persisted.test.test_styles.UnpickleMethodTests.test_instanceBuildingName* PyPy failures

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#9239]
|[<img alt="dansut's avatar" src="https://avatars.githubusercontent.com/u/100365?s=50" width="50" height="50">](https://github.com/dansut)| @dansut reported|
|-|-|
|Trac ID|trac#9239|
|Type|task|
|Created|2017-07-27 11:29:33Z|

The  
- twisted.persisted.test.test_styles.UnpickleMethodTests.test_instanceBuildingNamePresent
- twisted.persisted.test.test_styles.UnpickleMethodTests.test_instanceBuildingNameNotPresent
tests currently FAIL under PyPy 5.8.0 (both 2.7 and 3.5 compatible builds).

This happens because assigning bound methods to a variable works differently in PyPy currently: https://bitbucket.org/pypy/pypy/issues/2619/

Unsure of reason why this test is important though... ie. why the method returned from twisted.persisted.styles.unpickleMethod() needs to be `is not` the exact method passed in... if someone could explain to me that would be great :) 

<details><summary>Searchable metadata</summary>

```
trac-id__9239 9239
type__task task
reporter__dansut dansut
priority__normal normal
milestone__PyPy_support PyPy-support
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1501154973455803 1501154973455803
changetime__1594695277076905 1594695277076905
version__None None
owner__None None

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="dansut's avatar" src="https://avatars.githubusercontent.com/u/100365?s=50" width="50" height="50">](https://github.com/dansut)<a name="note_1"></a>|@dansut commented|
|-|-|

This was slight problem in PyPy (see above logged issue) and is now fixed in upstream PyPy now, awaiting PyPy5.8.1 release.

**Comment by twisted-trac** [comment]
|[<img alt="dansut's avatar" src="https://avatars.githubusercontent.com/u/100365?s=50" width="50" height="50">](https://github.com/dansut)<a name="note_2"></a>|@dansut commented|
|-|-|

As of the release of PyPy 5.9.0 this problem does not show for both the 2.7 and 3.5 builds of PyPy.

twisted.persisted.test.test_styles.UnpickleMethodTests now PASS

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_3"></a>|@rodrigc set status to `closed`|
|-|-|

Daniel pointed out that this is fixed upstream in Pypy 5.9.0
